state/factory/util.go

```diff
@@ -71,8 +72,16 @@ func calculateLogsBloom(ctx context.Context, receipts []*action.Receipt) bloom.B
 }
 
 // generateWorkingSetCacheKey generates hash key for workingset cache by hashing blockheader core and producer pubkey
-func generateWorkingSetCacheKey(blkHeader block.Header, producerAddr string) hash.Hash256 {
-	sum := append(blkHeader.SerializeCore(), []byte(producerAddr)...)
+func generateWorkingSetCacheKey(blkHeader block.Header, _ string) hash.Hash256 {
+	sum := byteutil.Uint64ToBytesBigEndian(blkHeader.Height())
+	h := blkHeader.PrevHash()
+	sum = append(sum, h[:]...)
+	h = blkHeader.TxRoot()
+	sum = append(sum, h[:]...)
+	h = blkHeader.DeltaStateDigest()
+	sum = append(sum, h[:]...)
+	h = blkHeader.ReceiptRoot()
+	sum = append(sum, h[:]...)
```

Comment:
height + prev hash + 3 roots => the uniqueness and correctness of the proposed block

Reply:
why is blkHeader.SerializeCore() replaced by height + prev hash + 3 roots?

Reply:
b/c header contains timestamp, which is changed when the next round proposer re-sign the block